src/dali/plugin/input/ods/rp2_input.py

```diff
@@ -0,0 +1,137 @@
+# Copyright 2023 jamesbaber1
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from typing import List, Optional, cast
+from datetime import date
+
+from rp2.logger import create_logger
+from rp2.ods_parser import parse_ods, open_ods
+from rp2.input_data import InputData
+from rp2.configuration import Configuration, MAX_DATE, MIN_DATE
+from rp2.in_transaction import InTransaction as RP2InTransaction
+from rp2.intra_transaction import IntraTransaction as RP2IntraTransaction
+from rp2.out_transaction import OutTransaction as RP2OutTransaction
+
+from dali.configuration import DEFAULT_CONFIGURATION, Keyword
+from dali.abstract_input_plugin import AbstractInputPlugin
+from dali.abstract_transaction import AbstractTransaction
+from rp2.abstract_country import AbstractCountry
+from dali.in_transaction import InTransaction
+from dali.intra_transaction import IntraTransaction
+from dali.out_transaction import OutTransaction
+
+
+class InputPlugin(AbstractInputPlugin):
+    __RP2_INPUT: str = "RP2 Input"
+
+    def __init__(
+        self,
+        configuration_path: str,
+        input_file: str,
+        native_fiat: Optional[str] = None,
+    ) -> None:
+        super().__init__(account_holder="", native_fiat=native_fiat)
+        self.__configuration_path: str = configuration_path
+        self.__input_file: str = input_file
+        self.__logger: logging.Logger = create_logger(self.__RP2_INPUT)
+        self.__country: AbstractCountry = DEFAULT_CONFIGURATION[Keyword.COUNTRY.value]
+        self.__from_date: date = MIN_DATE
+        self.__to_date: date = MAX_DATE
```

Comment:
Nit: same for `self.__from_date` and `self.__to_date`: you could just use `MIN_DATE` and `MAX_DATE` directly.